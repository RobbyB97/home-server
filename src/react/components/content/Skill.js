import React from 'react'

export default (props) => (
    <div className="skill">
        <div className="skill__title">
            {props.title}
        </div>
        
        {props.bits &&
            <div className="skill__bits">
                {props.bits.map((bit) => (
                    <p>{bit}</p>
                ))}
            </div>
        }
        
        <div className="skill__description">
            {props.description}
        </div>
    </div>
)