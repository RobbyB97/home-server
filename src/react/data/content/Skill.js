export const skill_frontEnd = {
    title: "Front End Development",
    bits: [
        'HTML5',
        'CSS3',
        'JS (ES6)',
        'SASS',
        {
            skill: 'React',
            list: [
                'Router',
                'Redux',
                'Jest'
            ]
        }
    ]
}

export const skill_backEnd = {
    title: "Back End Development",
    bits: [
        {
            skill: 'Node.JS',
            list: [
                'Express'
            ]
        },
        'PHP',
        'Heroku',
        {
            skill: 'SQL / NOSQL',
            list: [
                'Oracle SQL',
                'MongoDB'
            ]
        }
    ]
}

export const skill_programming = {
    title: "Programming",
    bits: [
        {
            skill: 'Workflow',
            list: [
                'NPM',
                'Webpack',
                'Gulp'
            ]
        },
        'Mobile First Development',
        'Test Driven Development',
        {
            skill: 'Languages',
            list: [
                'Python',
                'BASH'
            ]
        },
        {
            skill: 'Git',
            list: [
                'Github'
            ]
        }
    ]
}
