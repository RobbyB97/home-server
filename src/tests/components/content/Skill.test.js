import React from 'react'
import {shallow} from 'enzyme'

import Skill from '../../../react/components/content/Skill'


test('Render Skill with only a title', () => {
    const wrapper = shallow(<Skill title="Title!"/>)

    expect(wrapper).toMatchSnapshot()
})