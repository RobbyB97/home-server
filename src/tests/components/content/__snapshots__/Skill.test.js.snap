// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Skill with only a title 1`] = `
<div
  className="skill"
>
  <div
    className="skill__title"
  >
    Title!
  </div>
  <div
    className="skill__description"
  />
</div>
`;
